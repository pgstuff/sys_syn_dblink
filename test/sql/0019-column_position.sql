BEGIN;

CREATE EXTENSION sys_syn_dblink;

CREATE SCHEMA processor_data
        AUTHORIZATION postgres;

SELECT  dblink_connect('sys_syn_test', 'dbname=contrib_regression host=' ||
        quote_literal(split_part((SELECT pg_settings.setting FROM pg_settings WHERE pg_settings.name = 'unix_socket_directories'), ', ', 1)));
SELECT dblink_exec('sys_syn_test', 'BEGIN');

INSERT INTO sys_syn_dblink.in_groups_def VALUES ('sys_syn_dblink-test', 'in');
INSERT INTO sys_syn_dblink.out_groups_def VALUES ('sys_syn_dblink-test', 'out');
INSERT INTO sys_syn_dblink.put_groups_def VALUES ('put');

INSERT INTO sys_syn_dblink.put_column_transforms (
        priority,       out_group_id_like,      in_group_id_like,
        proc_table_id_like,       column_name_like,
        add_columns
) VALUES (
        150,            'out',                  'in',
        'test_table',           'test_table_text',
        ARRAY[
                ROW(    'id_first',    'smallint',     'Id',
                        $$1::smallint$$,
                        NULL,   'InColumnType',         TRUE,   ARRAY[]::TEXT[],
                        'Id'),
                ROW(    'id_last',     'text',         'Id',
                        $$'id'$$,
                        NULL,   'InColumnType',         FALSE,  ARRAY[]::TEXT[],
                        'Id'),
                ROW(    'attr',         'text',         'Attribute',
                        $$'attr'$$,
                        NULL,   'Here',                 FALSE,  ARRAY[]::TEXT[],
                        NULL)
        ]::sys_syn_dblink.create_put_column[]
);

SELECT sys_syn_dblink.proc_table_create (
        proc_schema     => 'processor_data',
        in_table_id     => 'test_table',
        out_group_id    => 'out',
        put_group_id    => 'put',
        dblink_connname => 'sys_syn_test');


SELECT * FROM processor_data.test_table_claim_1();

SELECT * FROM processor_data.test_table_pull_1();

SELECT  trans_id_in, delta_type, queue_priority, hold_updated, prior_hold_reason_count, prior_hold_reason_id, prior_hold_reason_text, id, attributes, no_diff
FROM    processor_data.test_table_processing_1
ORDER BY id, attributes;

SELECT * FROM processor_data.test_table_process_1();

SELECT  *
FROM    test_table
ORDER BY test_table_id, test_table_text;

SELECT  hold_reason_id, hold_reason_text, queue_priority
FROM    processor_data.test_table_processed_1
ORDER BY id;


SELECT dblink_exec('sys_syn_test', 'ROLLBACK');
SELECT dblink_disconnect('sys_syn_test');

ROLLBACK;
