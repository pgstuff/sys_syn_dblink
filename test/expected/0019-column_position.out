BEGIN;
CREATE EXTENSION sys_syn_dblink;
CREATE SCHEMA processor_data
        AUTHORIZATION postgres;
SELECT  dblink_connect('sys_syn_test', 'dbname=contrib_regression host=' ||
        quote_literal(split_part((SELECT pg_settings.setting FROM pg_settings WHERE pg_settings.name = 'unix_socket_directories'), ', ', 1)) ||
        'port=' || current_setting('port'));
 dblink_connect 
----------------
 OK
(1 row)

SELECT dblink_exec('sys_syn_test', 'BEGIN');
 dblink_exec 
-------------
 BEGIN
(1 row)

INSERT INTO sys_syn_dblink.in_groups_def VALUES ('sys_syn_dblink-test', 'in');
INSERT INTO sys_syn_dblink.out_groups_def VALUES ('sys_syn_dblink-test', 'out');
INSERT INTO sys_syn_dblink.put_groups_def VALUES ('put');
INSERT INTO sys_syn_dblink.put_column_transforms (
        priority,       out_group_id_like,      in_group_id_like,
        proc_table_id_like,       column_name_like,
        add_columns
) VALUES (
        150,            'out',                  'in',
        'test_table',           'test_table_text',
        ARRAY[
                ROW(    'id_first',    'smallint',     'Id',
                        $$1::smallint$$,
                        NULL,   'InColumnType',         TRUE,   ARRAY[]::TEXT[],
                        'Id'),
                ROW(    'id_last',     'text',         'Id',
                        $$'id'$$,
                        NULL,   'InColumnType',         FALSE,  ARRAY[]::TEXT[],
                        'Id'),
                ROW(    'attr',         'text',         'Attribute',
                        $$'attr'$$,
                        NULL,   'Here',                 FALSE,  ARRAY[]::TEXT[],
                        NULL)
        ]::sys_syn_dblink.create_put_column[]
);
SELECT sys_syn_dblink.proc_table_create (
        proc_schema     => 'processor_data',
        in_table_id     => 'test_table',
        out_group_id    => 'out',
        put_group_id    => 'put',
        dblink_connname => 'sys_syn_test');
 proc_table_create 
-------------------
 
(1 row)

SELECT * FROM processor_data.test_table_claim_1();
 test_table_claim_1 
--------------------
 t
(1 row)

SELECT * FROM processor_data.test_table_pull_1();
 test_table_pull_1 
-------------------
 t
(1 row)

SELECT  trans_id_in, delta_type, queue_priority, hold_updated, prior_hold_reason_count, prior_hold_reason_id, prior_hold_reason_text, id, attributes, no_diff
FROM    processor_data.test_table_processing_1
ORDER BY id, attributes;
 trans_id_in | delta_type | queue_priority | hold_updated | prior_hold_reason_count | prior_hold_reason_id | prior_hold_reason_text | id  |   attributes    | no_diff 
-------------+------------+----------------+--------------+-------------------------+----------------------+------------------------+-----+-----------------+---------
           1 | Add        |                |              |                         |                      |                        | (1) | ("test_data 1") | 
           1 | Add        |                |              |                         |                      |                        | (2) | ("test_data 2") | 
           1 | Add        |                |              |                         |                      |                        | (3) | ("test_data 3") | 
(3 rows)

SELECT * FROM processor_data.test_table_process_1();
 test_table_process_1 
----------------------
 t
(1 row)

SELECT  *
FROM    test_table
ORDER BY test_table_id, test_table_text;
 id_first | test_table_id | id_last | test_table_text | attr 
----------+---------------+---------+-----------------+------
        1 |             1 | id      | test_data 1     | attr
        1 |             2 | id      | test_data 2     | attr
        1 |             3 | id      | test_data 3     | attr
(3 rows)

SELECT  hold_reason_id, hold_reason_text, queue_priority
FROM    processor_data.test_table_processed_1
ORDER BY id;
 hold_reason_id | hold_reason_text | queue_priority 
----------------+------------------+----------------
                |                  |               
                |                  |               
                |                  |               
(3 rows)

SELECT dblink_exec('sys_syn_test', 'ROLLBACK');
 dblink_exec 
-------------
 ROLLBACK
(1 row)

SELECT dblink_disconnect('sys_syn_test');
 dblink_disconnect 
-------------------
 OK
(1 row)

ROLLBACK;
